### PR TITLE
fix(e2e): upgrade test fixes

### PR DESCRIPTION
**Pull request by nojnhuh**
**Description**:
These changes fix current issues with the upgrade e2e test:

- Remove a check that the `kubectl` executable is present before running
  the upgrade e2e test. Because CRDs are now upgraded with hooks in the
  Helm chart instead of invoking `kubectl` in the test, this is no
  longer necessary.

- Remove the step to reinstall the SMI policy resources, since that is
  no longer necessary with the new CRD management.

- Restart the injected Pods to handle incompatible certificate common
  name formats between the old and new OSM installations.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [X] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? No
    -   Did you notify the maintainers and provide attribution?

1. Is this a breaking change? No

### CODE REVIEW

**Review by snehachhabria — APPROVED** [review]

File: tests/e2e/e2e_upgrade_test.go [hunk 1]
```diff
@@ -1,7 +1,6 @@
 package e2e
 
 import (
-	"os/exec"
```
Comment (snehachhabria) [inline tests/e2e/e2e_upgrade_test.go:4]: We can remove the `// Re-deploy policies` code in line 218 as a part of this PR too

**Review by ksubrmnn — APPROVED** [review]
Would it make sense to have a flag in `osm upgrade` to handle restarting these deployments automatically?

**Comment by nojnhuh** [comment]
> Would it make sense to have a flag in `osm upgrade` to handle restarting these deployments automatically?

I suspect that would be possible, but might be tricky in certain cases. Like if there's a Pod in the mesh not backed by a controller, can we figure out what was originally originally created vs. what was added by a mutating webhook? In that case would recreating the Pod in the mesh from its current spec mean the osm-injector would add a second sidecar? I'm sure other non-trivial things like that would pop up.

My gut tells me it's probably not worth it. If I were a user that saw that flag, I wouldn't use it just in case it does something destructive and just restart my workloads myself since I would know best how to do that.

**Review by snehachhabria — APPROVED** [review]
thanks for fixing!

**Pull request merged**